src/js/http_client.js

```diff
@@ -29,7 +29,24 @@ function ClientRequest(options, cb) {
   var port = options.port = options.port || 80;
   var host = options.host = options.hostname || options.host || '127.0.0.1';
   var method = options.method || 'GET';
-  var path = options.path || '/';
+  this.path = options.path || '/';
+
+  var methodIsString = (typeof method === 'string');
+  if (methodIsString && method) {
```

Comment:
I think we just defines all methods and check by `methods[method]` here.

Reply:
Actually it is possible to use a custom method rather than most common methods like GET, POST in http request.
#ref: https://tools.ietf.org/html/rfc2616#section-5.1.1

Reply:
I think we don't need to support the whole context-free grammar for `Method`, a reduced CFG is:

```
Method         = "OPTIONS"                ; Section 9.2
                      | "GET"                          ; Section 9.3
                      | "HEAD"                       ; Section 9.4
                      | "POST"                       ; Section 9.5
                      | "PUT"                         ; Section 9.6
                      | "DELETE"                   ; Section 9.7
                      | "TRACE"                     ; Section 9.8
                      | "CONNECT"               ; Section 9.9
```

> An origin server SHOULD return the status code 405 (Method Not Allowed) if the method is known by the origin server but not allowed for the requested resource, and 501 (Not Implemented) if the method is unrecognized or not implemented by the origin server.

As the above description, we could respond a 501 for those methods is invalid.

Reply:
Hmm, really think there are not too much developers to use the custom feature. But let's support it, that's reasonable than me.